vizier/backend/src/info/vizierdb/commands/plot/ScatterPlot.scala

```diff
@@ -0,0 +1,296 @@
+    )),
+    StringParameter(id = PARAM_ARTIFACT, name = "Output Artifact (blank to show only)", required = false)
+  )
+  override def title(arguments: Arguments): String = 
+    "PLOT "+arguments.getList(PARAM_SERIES).map { series =>
+      series.pretty(PARAM_DATASET)
+    }.toSet.mkString(", ")
+
+  override def format(arguments: Arguments): String = 
+    "PLOT "+arguments.getList(PARAM_SERIES).map { series =>
+      f"${series.pretty(PARAM_DATASET)}.{${series.pretty(PARAM_X)}, ${series.pretty(PARAM_Y)}}${series.getOpt[String](PARAM_LABEL).map { " AS " + _ }.getOrElse("")}"
+    }.mkString("\n     ")
+
+  override def process(arguments: Arguments, context: ExecutionContext): Unit = 
+  {
+    //Variables to store axes titles
+    var xTitle = ""
+    var yTitle = ""
+
+    //Variables to hold the x and y values
+    var xValues = scala.collection.mutable.ListBuffer[Double]()
+    var yValues = scala.collection.mutable.ListBuffer[Double]()
+
+    //Variables to hold all of the dataset names, y columns, x columns, and combinations of them along with boolean values that can be used to identify if those naming conventions create all unique names
+    var datasetNames = scala.collection.mutable.Set[String]()
+    var datasetUnique = true
+    var yColumns = scala.collection.mutable.Set[String]()
+    var yUnique = true
+    var xColumns = scala.collection.mutable.Set[String]()
+    var xUnique = true
+    var datasetAndY = scala.collection.mutable.Set[String]()
+    var datasetAndYUnique = true
+    var datasetAndX = scala.collection.mutable.Set[String]()
+    var datasetAndXUnique = true
+
+    // Figure out if we are being asked to emit a named artifact
+    // Store the result in an option-type
+    val artifactName = arguments.getOpt[String](PARAM_ARTIFACT)
+                                .flatMap { case "" => None 
+                                           case x => Some(x) }
+
+    val nameComponents:Seq[(String, String, String)] = arguments.getList(PARAM_SERIES).map {
+	series => 
+		val datasetName = series.get[String](PARAM_DATASET)
+		val xColIdx = series.get[Int](PARAM_X)
+        	val yColIdx = series.get[Int](PARAM_Y)
+		var dataset = context.dataframe(datasetName)
+		val xCol = dataset.columns(xColIdx)
+		val yCol = dataset.columns(yColIdx)
+		(datasetName, yCol, xCol)
+    }
+
+    //Variables to hold the number of unique names for each naming convention
+    val numberOfUniqueDatasetNames = nameComponents.map { c => c._1 }.toSet.size
+    val numberOfUniqueYNames = nameComponents.map { c => c._2 }.toSet.size
+    val numberOfUniqueXNames = nameComponents.map { c => c._3 }.toSet.size
+    val numberOfUniqueDatasetandYNames = nameComponents.map { c => (c._1 + c._2) }.toSet.size
+    val numberOfUniqueDatasetandXNames = nameComponents.map { c => (c._1 + c._3) }.toSet.size
+
+    //lambda function to generate a series label based on uniqueness
+    val makeLabel = (datasetName: String, yCol: String, xCol: String) =>
```

Comment:
Move the `(datasetName: ... ) =>` *inside* the if statement blocks please.  This will define one function for each if statement outcome, rather than one statement with an if block contained within.  It's not a huge performance difference here, but it's a slightly cleaner structure.